tests/snappi_tests/utils/metrics.py

```diff
@@ -0,0 +1,116 @@
+"""
+This file defines the classes receiving metrics from snappi tests and processing them.
+"""
+import logging
+import json
+import datetime
+import time
+
+from copy import deepcopy
+from pprint import pprint
+from typing import Dict, Final, List, Union
+
+# Only certain labels are allowed
+METRIC_LABEL_TEST_TESTBED: Final[str] = "test.testbed"
+METRIC_LABEL_TEST_BUILD: Final[str] = "test.os.version"
+METRIC_LABEL_TEST_CASE: Final[str] = "test.testcase"
+METRIC_LABEL_TEST_FILE: Final[str] = "test.test_file"
+METRIC_LABEL_TEST_JOBID: Final[str] = "test.job_id"
+METRIC_LABEL_DEVICE_ID: Final[str] = "device.id"
+METRIC_LABEL_DEVICE_PORT_ID: Final[str] = "device.port.id"
+METRIC_LABEL_DEVICE_PSU_ID: Final[str] = "device.psu.id"
+METRIC_LABEL_DEVICE_QUEUE_ID: Final[str] = "device.queue.id"
+METRIC_LABEL_DEVICE_SENSOR_ID: Final[str] = "device.sensor.id"
```

Comment:
I believe we are missing many device related fields here. Check the single switch stress design doc, you will find the gaps there.

Reply:
You mean "priority_group" and "buffer_pool"? Anything else?

Reply:
and actually more, for example, in PSU, we have PSU model, serial, HW Rev.

going to a device and run the command will help you getting these things more well-understood.